drape_frontend/stylist.cpp

```diff
@@ -213,16 +213,20 @@ class Aggregator
       m2::RectD const r = m_f.GetLimitRect(m_zoomLevel);
       // Raw areas' size range is about (1e-11, 3000).
       double const areaSize = r.SizeX() * r.SizeY();
-      // Use log2() to have more precision distinguishing smaller areas.
-      double const areaSizeCompact = std::log2(areaSize);
       // Compacted range is approx (-37;13).
-      double constexpr minSize = -37,
-                       maxSize = 13,
-                       stretchFactor = kDepthRangeBgBySize / (maxSize - minSize);
-      // Adjust the range to fit into [kBaseDepthBgBySize;kBaseDepthBgTop).
-      m_areaDepth = kBaseDepthBgBySize + (maxSize - areaSizeCompact) * stretchFactor;
+      double constexpr kMinSize = -37,
+                       kMaxSize = 13,
+                       kStretchFactor = kDepthRangeBgBySize / (kMaxSize - kMinSize);
+
+      // Use log2() to have more precision distinguishing smaller areas.
+      /// @todo We still can get here with areaSize == 0.
+      double const areaSizeCompact = std::max(kMinSize, (areaSize > 0) ? std::log2(areaSize) : kMinSize);
```

Comment:
why std::max()?
the expression shouldn't result in a value lower than kMinSize anyway

and if it ever will be lower then it'll be caught by an assert below

Reply:
To conform next m_areaDepth calculation, areaSizeCompact should fit: minSize <= areaSizeCompact <= maxSize
Check only lower bound here.

Reply:
if e.g. 1.0E-12 areaSize ever happens then we'd want to review kMinSize, so add an assert please or maybe a log even (it should be something very rare anyways)

Reply:
Can we assert and fix these areas in generator, to avoid runtime overhead? Maybe as a TODO issue for the future.

Reply:
Of course, I will check a generator. This patch is for existing data files, and (probably) should fix user's problem https://github.com/organicmaps/organicmaps/issues/6009

There is no overhead with 2 comparisons.

Reply:
The best would be of course to log a smallest non-zero area size in the generator, so that we can adjust kMinSize if needed.

Reply:
In general, <1E-5 precision is not supported, so all valid areas should be bigger than 1E-11. I observed 0 for invalid areas. Also, I suspect that uint -> double coordinates decoding may? produce calculation errors, so we may? get some small <1E-11 values instead of 0. I didn't observe it, but don't see a problem to check the lower bound here.